### ISSUE TITLE
num_queries parameter

### ISSUE THREAD

**Issue by ThomasDougherty** [facebookresearch/detr#126]
If we wanted to drop our `num_queries` parameter to say 30, what other parameters need to change and how? I've read that `eos_coef` needs to be adjusted when `num_queries` changes but by how much specifically? Any other things worth noting when playing with `num_queries` ? Insight would be greatly appreciated.

**Comment by alcinos** [comment]
Hi @ThomasDougherty 
Thanks for your interest in DETR
Adjusting this properly is a bit of a tricky question, it depends on several things including possibly the number of classes you have.
May I ask why you are looking to decrease it? In general, you'll get better results with a higher query count since they will be able to specialize more efficiently.
If you want to give 30 queries a go anyway, I'd recommend trying first with the default params, then see if the effect of varying the eos_coef (say by 2x or 0.5x)

Best of luck.

**Comment by ThomasDougherty** [comment]
The max number of objects I have in a single image is 17. Some of my classes have very thin bounding boxes (~3 pixels wide) and the DETR model is struggling with generating bounding boxes that are thin enough for a good IOU. I was hoping decreasing the queries would allow the model to focus on bounding boxes thin enough for these classes. Also will try dropping the learning rate significantly during training.

**Comment by fmassa** [comment]
Hi,

DETR might struggle to accurately localize very thin objects because we use feature maps which have been downsampled by a factor of 32.

I believe we have answered your question so I'm closing this issue, but let us know how your experiments go and let us know if you have any further questions.

**Comment by alcinos** [comment]
I concur with @fmassa, thin objects are going to be an issue regardless of the number of queries. I'd advice switching to DC5 if you can afford it or mb try playing a bit with the giou loss coeff.

**Comment by RicardoQuinteiro** [comment]
This might actually be a dumb question but can I use the weights that you pretrained with COCO dataset while lowering the num_queries parameter? The thing is I have a dataset whose goal is to detect only one kind of object but most of the images don't have any and the maximum number of objects I have in one image is 3 or 4 so having num_queries=100 seems to be a bit of an overkill. Also, the dataset is not that big (~7500 images) either so I guess it would be better to use transfer learning.

**Comment by fmassa** [comment]
You could try lowering the number of queries, but I would first start with 100 and fine tune so that you have a baseline

**Comment by RicardoQuinteiro** [comment]
Ok, thank you. I have another question. Since I don't have many examples I want the misclassification of the objects to have a higher cost. This can be done by lowering the eos_coef, right? Should I also try to adjust the loss parameters?

**Comment by fmassa** [comment]
I would say it would be better to try a few different things to see what works best in your case. I would start with the current default parameters, and then change from there.

**Comment by ahmadrezabaqerzade** [comment]
model.query_embed = nn.Embedding(num_querie you want, 256)
